public/js/functions.js

```diff
@@ -470,6 +492,95 @@ function concat(list) {
 	return Array.prototype.concat.apply([], list)
 }
 
+//list of list to list
+function uconcat(list) {
+	return Unique(Array.prototype.concat.apply([], list));
+}
+
+function formatDate(list,format) {
+	if(Array.isArray(list)) {
+        return list.map(function(item){
+        	return dateFormat(item,format);
+        });
+    }    
+    return dateFormat(list,format); 
+}
+
+function addDays(list,days,businessDays) {
+	businessDays = String(businessDays || true).toLowerCase() == "true"
+
+	if(Array.isArray(list)) {
+        return list.map(function(item){
+        	return setDays(item);
+        });
+    } 
+
+    function businessDay(date) {    	
+    	switch(date.getDay()) {
+    		case 0: return (days > 0) ? date.setDate(date.getDate() + 1) : date.setDate(date.getDate() - 2);
+    		case 6: return (days > 0) ? date.setDate(date.getDate() + 2) : date.setDate(date.getDate() - 1);
+    		default:return date;
+    	}    	
+    }
+
+    function setDays(item) {
+    	var date = new Date(dateFormat(item));
+    	date.setDate(date.getDate() + days);
+    	if(businessDays) return businessDay(date);
+    	return date;
+    }
+
+    return setDays(list);
+}
+
+function format(list) {  
+	if(Array.isArray(list)) {
+        return list.map(function(item){
+        	return item.toLocaleString();
+        });
+    }
+    return list.toLocaleString();      
+}
+
+function coalesce(list){
+	if(Array.isArray(list)){
+		var ret = [];
+		list.forEach(function(item){
+			if(!(item === null || item === undefined)) ret.push(item);
+		});
+		return ret;
+	}		
+	return list;	
+}
+
+function numbers(list){
+	if(Array.isArray(list)){
+		var ret = [];
+		list.forEach(function(item){
+			if(!(item === null || item === undefined || isNaN(item))) ret.push(item);
+		});
+		return ret;
+	}		
+	return list;	
+}
+
+function isnull(x, val){
+	if(!(x === null || x === undefined || isNaN(x))) return x;	
+	return val;	
+}
+
+function returns(list){
+	if(Array.isArray(list)){
+		var ret = [];
+		list.forEach(function(item,indx,ary){
+			if(indx==0||item==0) return ret.push(0);
```

Comment:
Remove this ... if you need to do this make the list a qube
e.g.

Index[] = list
Return[#Index==0] = 0
Return = Index / Index[Prev(Index)] - 1